backtracking/minimax.cpp

```diff
@@ -1,27 +1,32 @@
+/**
+ * @file
+ * @brief brief one line description here
+ */
+#include <algorithm>
 #include <cmath>
 #include <iostream>
 #include <vector>
 
-using std::cout;
-using std::endl;
-using std::max;
-using std::min;
-using std::vector;
-
-int minimax(int depth, int node_index, bool is_max, vector<int> scores,
+int minimax(int depth, int node_index, bool is_max, std::vector<int> scores,
```

Comment:
* Missing function documentation
* Pass class variables as const references if they are not being modified or as pointers of they are being modified